spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -156,15 +166,30 @@ object DistriOptimizer {
         models, true)(
         (data, modelIter) => {
           val cached = modelIter.next()
+          val executorId = SparkEnv.get.executorId
+          val parameters = AllReduceParameterManager.get(pid, executorId)
           val syWStart = System.nanoTime()
           /*
             Note: All models in `cached` share the same storage for weights, so we only need to
             copy the weights from parameter server into the first model's weights.
            */
-          val weightsResult = parameters.getWeights(cached.modelWeights.head)
+          var weightsResult: FutureResult[Int] = null
+          if (partitionNum == Engine.nodeNumber()) {
+            weightsResult = parameters.getWeights(cached.modelWeights.head)
+          } else {
+            AllReduceParameterManager.synchronized {
+              if (!parameters.job1Start) {
+                weightsResult = parameters.getWeights(cached.modelWeights.head)
+                weightsResult.waitResult()
+                parameters.job1Start = true
```

Comment:
It's used to make sure in each executor, get weight is only executed once, in master code, we don't need worry about get weights call more than 1 as in each node there is only 1 task.